src/phy/mod.rs

```diff
@@ -265,3 +265,8 @@ pub trait TxToken {
     fn consume<R, F>(self, timestamp: Instant, len: usize, f: F) -> Result<R>
         where F: FnOnce(&mut [u8]) -> Result<R>;
 }
+
+
+/// Ethernet header length to be added to the MTU, since the header is not counted into MTU
+/// 6 bytes src addr, 6 bytes dst addr, 2 bytes EthType, (optional) 4 bytes IEEE 802.1Q Header
+pub const ETHERNET_HAEDER_MAX_LEN: usize = 18;
```

Comment:
This doesn't belong in this pull request. Please submit it separately and we'll have a discussion about it; in particular, you're counting 802.1Q, but not QinQ or QinQinQ.

Reply:
This is needed to get the fragmentation to work. In Linux MTU refers to the max length of the ethernet payload. But smoltcp counts MTU as max length of the ethernet frame. Hence without this added ethernet length, all fragmented packets are truncated. I am happy to fix is more meaningfully later, but without this fix the fragmentation reassembly doesn't work.

Reply:
Can you extract it into a separate PR that we merge before this one?

Reply:
See #237